addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/dto/HueStateColorBulb.java

```diff
@@ -78,17 +78,65 @@ public HueStateColorBulb(HSBType hsb) {
      * Converts this HueState to a HSBType
      */
     public HSBType toHSBType() {
-        int bri = this.bri * 100 / MAX_BRI;
-        int sat = this.sat * 100 / MAX_SAT;
-        int hue = this.hue * 360 / MAX_HUE;
-
-        if (!this.on) {
-            bri = 0;
-        }
-        if (colormode == ColorMode.ct) {
-            sat = 0;
+        if (colormode == ColorMode.xy) {
+            int i;
+            double d = this.xy[0];
+            d = this.xy[1];
+            double Y = ((double) this.bri) / 100.0d;
+            double X = (Y / this.xy[1]) * this.xy[0];
+            double Z = (Y / this.xy[1]) * ((1.0d - this.xy[0]) - this.xy[1]);
+            int R = (int) (Math.abs(((1.4628067016601562d * X) - (0.18406230211257935d * Y)) - (0.2743605971336365d * Z)) * 255.0d);
+            int G = (int) (Math.abs((((-X) * 0.5217933058738708d) + (1.4472380876541138d * Y)) + (0.06772270053625107d * Z)) * 255.0d);
+            int B = (int) (Math.abs(((0.03493420034646988d * X) - (0.09689299762248993d * Y)) + (1.288409948348999d * Z)) * 255.0d);
+            if (R < G) {
+                i = R;
+            } else {
+                i = G;
+            }
+            double mi = (double) i;
+            if (mi >= ((double) B)) {
+                mi = (double) B;
+            }
+            double ma = (double) (R > G ? R : G);
+            if (ma <= ((double) B)) {
+                ma = (double) B;
+            }
+            double delta = ma - mi;
+            if (ma <= 0.0d) {
+                return new HSBType(new DecimalType(0), new PercentType(100), new PercentType((this.bri * 100) / MAX_BRI));
+            }
+            double h;
+            if (((double) R) >= ma) {
+                h = ((double) (G - B)) / delta;
+            } else if (((double) G) >= ma) {
+                h = 2.0d + (((double) (B - R)) / delta);
+            } else {
+                h = 4.0d + (((double) (R - G)) / delta);
+            }
+            h *= 60.0d;
+            if (h < 0.0d) {
+                h += 360.0d;
+            }
+            double hue_sat = Math.floor((delta / ma) * 254.0d);
```

Comment:
Can this be hueSat to follow naming standards, apply everywhere :-)

Reply:
Yes this can be changed of course